### DISCUSSION TITLE
Ability to create identities in CandidUI

### DISCUSSION THREAD

**Discussion by ocluf** [dfinity/sdk#2910] [Feature Requests]
CandidUI is fantastic for quickly checking out if the functions of my canister work as expected. Passing a recordtype on the command line can become incredibly bothersome when you need to do it in the candid textual format without any autocompletion, errors, or syntax highlighting. 

However there also hasn't been a single project I did where I didn't use the `caller` of the call somehow. There aren't many serious projects that don't have some form of authentication. In these cases I find myself still needing to either write repl scripts or constantly changing the code to remove authentication checks

It would be very useful if you would be able to select multiple principals to call from in the CandidUI. They don't need to be pulled form dfx or anything, just ephemeral principals would be a huge help. 

I know there was a community project that made something like this at well, but I couldn't find it and most people will simply click the link that comes up when you deploy a canister.

**Comment by viviveevee** [comment]
I agree this would be a useful feature. I'll let the candid team know.

In case anyone wants to take this into their own hands, [here](https://github.com/dfinity/candid/tree/master/tools/ui) is the candid UI source code.

> I know there was a community project that made something like this at well, but I couldn't find it

I think you mean [Candid+](https://devpost.com/software/candid-7ko9uz)

**Comment by chenyan-dfinity** [comment]
Thanks for the report. We will put it in the backlog, but may not be able to get to it very soon. Community contributions are always welcome :) 

icscan.io also implements similar features with identity login.

**Comment by Web3NL** [comment]
@ocluf you can login with II now.
